app/org/sagebionetworks/bridge/services/ParticipantService.java

```diff
@@ -233,13 +233,25 @@ public void createSmsRegistration(Study study, String userId) {
         notificationsService.createRegistration(study.getStudyIdentifier(), criteriaContext, registration);
     }
 
-    public StudyParticipant getParticipant(Study study, AccountId accountId, boolean includeHistory) {
+    public StudyParticipant getParticipant(Study study, String userId, boolean includeHistory) {
+        // This parse method correctly deserializes formats such as externalId:XXXXXXXX.
+        AccountId accountId = BridgeUtils.parseAccountId(study.getIdentifier(), userId);
         Account account = getAccountThrowingException(accountId);
         return getParticipant(study, account, includeHistory);
     }
-
-    public StudyParticipant getParticipant(Study study, String id, boolean includeHistory) {
-        return getParticipant(study, AccountId.forId(study.getIdentifier(),  id), includeHistory);
+    
+    public StudyParticipant getSelfParticipant(Study study, CriteriaContext context, boolean includeHistory) {
+        AccountId accountId = AccountId.forId(study.getIdentifier(),  context.getUserId());
```

Comment:
Where are the unit tests for getSelfParticipant?

Reply:
Added